In [34]:
import sklearn
import pandas as pd
import numpy as np

### Importantar o dataset
#### O arquivo precisa está na mesma pasta do notebook

In [35]:
dfs = pd.read_csv('dataset_suap_tecnico_integrado_monteiro.csv', sep=';'  , engine='python')

## Análise do conjunto de dados

### Analisando os primeiros valores e o cabeçalho

In [36]:
dfs.head()

ID     Matricula  Ano letivo  Periodo Letivo Periodo na Matriz  \
0  1828  201615500018        2016               1                 1   
1  1829  201615500018        2016               1                 1   
2  1830  201615500018        2016               1                 1   
3  1831  201615500018        2016               1                 1   
4  1832  201615500018        2016               1                 1   

  Codigo do Componente Descricao do Componente  CH do Componente Nota  \
0             TIN.0003                   Artes                67   76   
1             TIN.0022              Biologia I                67   75   
2             TIN.0112       Educacao Fisica I                67   92   
3             TIN.0086             Filosofia I                33   88   
4             TIN.0125                Fisica I                67   89   

   % Frequencia  ... Faixa de Renda (SISTEC)  Forma de Ingresso  \
0           100  ...              0<RFP<=0,5               PSCT   
1            99  ...              0<RFP<=0,5               PSCT   
2           100  ...              0<RFP<=0,5               PSCT   
3            98  ...              0<RFP<=0,5               PSCT   
4           100  ...              0<RFP<=0,5               PSCT   

    Naturalidade                           Nome da Escola de Origem  \
0  Monteiro - PB  E.M.E.F.II Professora Maria do Socorro Aragao ...   
1  Monteiro - PB  E.M.E.F.II Professora Maria do Socorro Aragao ...   
2  Monteiro - PB  E.M.E.F.II Professora Maria do Socorro Aragao ...   
3  Monteiro - PB  E.M.E.F.II Professora Maria do Socorro Aragao ...   
4  Monteiro - PB  E.M.E.F.II Professora Maria do Socorro Aragao ...   

  Periodo de Ingresso Sexo Situacao no Curso Tipo Escola de Origem  \
0                   1    F         Concluido               Publica   
1                   1    F         Concluido               Publica   
2                   1    F         Concluido               Publica   
3                   1    F         Concluido               Publica   
4                   1    F         Concluido               Publica   

  Ultimo Ano/Periodo Letivo Zona Residencial do Aluno  
0                    2018/1                     Rural  
1                    2018/1                     Rural  
2                    2018/1                     Rural  
3                    2018/1                     Rural  
4                    2018/1                     Rural  

[5 rows x 28 columns]

In [37]:
dfs_cadastro_geral = dfs[['Matricula','Curso','Ano de Ingresso','Cor/Raca','Cota MEC','Cota SISTEC','Data de Nascimento','Faixa de Renda (SISTEC)','Nome da Escola de Origem','Sexo','Tipo Escola de Origem','Zona Residencial do Aluno','Situacao no Diario','Situacao no Curso','Codigo do Componente']]

In [38]:
dfs_cadastro_geral = dfs_cadastro_geral.sort_values(by="Matricula")

In [39]:
dfs_cadastro_geral[dfs_cadastro_geral.Matricula == 201825130017]

Empty DataFrame
Columns: [Matricula, Curso, Ano de Ingresso, Cor/Raca, Cota MEC, Cota SISTEC, Data de Nascimento, Faixa de Renda (SISTEC), Nome da Escola de Origem, Sexo, Tipo Escola de Origem, Zona Residencial do Aluno, Situacao no Diario, Situacao no Curso, Codigo do Componente]
Index: []

In [40]:
dfs_cadastro_geral = dfs_cadastro_geral[(~dfs_cadastro_geral['Situacao no Diario'].str.contains("Cancelado"))]

In [41]:
dfs_cadastro_geral.drop_duplicates(subset ='Matricula', 
                     keep='last', inplace = True) 

In [42]:
searchfor2 = [
'TIN.0056',
'TIN.0113',
'TIN.0087',
'TIN.0033',
'TIN.0027',
'TIN.0028',
'TIN.0030',
'TIN.0021',
'TIN.0161',
'TIN.0024',
'TIN.0091',
'TIN.0043',
'TIN.0008',
'TIN.0700',
'TIN.0132',
'TIN.0673',
'TIN.0675',
'TIN.0891',
'TIN.0893'
]

In [43]:
searchfor_cod_disciplinas1 = [
'TIN.0022',
'TIN.0138',
'TIN.0128',
'TIN.0112',
'TIN.0086',
'TIN.0125',
'TIN.0194',
'TIN.0352',
'TIN.0559',
'TIN.0351',
'TIN.0018',
'TIN.0001',
'TIN.0124',
'TIN.0090',
'TIN.0322',
'TIN.0014',
'TIN.0133',
'TIN.0150',
'TIN.0253'
]

dfs_cadastro_geral_disciplinas_concluidas_segundo_ano = dfs_cadastro_geral[(dfs_cadastro_geral['Codigo do Componente'].str.contains('|'.join(searchfor2)))]

In [44]:
dfs_cadastro_geral_disciplinas_concluidas_primeiro_ano = dfs_cadastro_geral[(dfs_cadastro_geral['Codigo do Componente'].str.contains('|'.join(searchfor_cod_disciplinas1)))]

In [45]:
dfs_cadastro_geral_disciplinas_concluidas_primeiro_ano.shape

(232, 15)

In [46]:
dfs_cadastro_geral_disciplinas_concluidas_primeiro_ano.head(50)

Matricula                                              Curso  \
324   201615500003  550 - Tecnico em Manutencao e Suporte em Infor...   
273   201615500007  550 - Tecnico em Manutencao e Suporte em Infor...   
2176  201615500008  550 - Tecnico em Manutencao e Suporte em Infor...   
206   201615500009  550 - Tecnico em Manutencao e Suporte em Infor...   
1668  201615500014  550 - Tecnico em Manutencao e Suporte em Infor...   
1378  201615500016  550 - Tecnico em Manutencao e Suporte em Infor...   
1230  201615500019  550 - Tecnico em Manutencao e Suporte em Infor...   
1280  201615500024  550 - Tecnico em Manutencao e Suporte em Infor...   
1497  201615500026  550 - Tecnico em Manutencao e Suporte em Infor...   
768   201615500030  550 - Tecnico em Manutencao e Suporte em Infor...   
856   201615500033  550 - Tecnico em Manutencao e Suporte em Infor...   
105   201615500034  550 - Tecnico em Manutencao e Suporte em Infor...   
1540  201615500038  550 - Tecnico em Manutencao e Suporte em Infor...   
722   201615500039  550 - Tecnico em Manutencao e Suporte em Infor...   
406   201615500041  550 - Tecnico em Manutencao e Suporte em Infor...   
2966  201615510013  551 - Tecnico em Instrumento Musical Integrado...   
3325  201615510018  551 - Tecnico em Instrumento Musical Integrado...   
3629  201615510021  551 - Tecnico em Instrumento Musical Integrado...   
2613  201615510023  551 - Tecnico em Instrumento Musical Integrado...   
3346  201615510025  551 - Tecnico em Instrumento Musical Integrado...   
3494  201615510028  551 - Tecnico em Instrumento Musical Integrado...   
3920  201615510035  551 - Tecnico em Instrumento Musical Integrado...   
2577  201615510038  551 - Tecnico em Instrumento Musical Integrado...   
3677  201615510039  551 - Tecnico em Instrumento Musical Integrado...   
2886  201615510042  551 - Tecnico em Instrumento Musical Integrado...   
5483  201715500002  550 - Tecnico em Manutencao e Suporte em Infor...   
5403  201715500008  550 - Tecnico em Manutencao e Suporte em Infor...   
5376  201715500010  550 - Tecnico em Manutencao e Suporte em Infor...   
4832  201715500013  550 - Tecnico em Manutencao e Suporte em Infor...   
5042  201715500014  550 - Tecnico em Manutencao e Suporte em Infor...   
4729  201715500021  550 - Tecnico em Manutencao e Suporte em Infor...   
4620  201715500022  550 - Tecnico em Manutencao e Suporte em Infor...   
4195  201715500028  550 - Tecnico em Manutencao e Suporte em Infor...   
4264  201715500030  550 - Tecnico em Manutencao e Suporte em Infor...   
5289  201715500037  550 - Tecnico em Manutencao e Suporte em Infor...   
5441  201715500040  550 - Tecnico em Manutencao e Suporte em Infor...   
6124  201715510006  551 - Tecnico em Instrumento Musical Integrado...   
6181  201715510007  551 - Tecnico em Instrumento Musical Integrado...   
6225  201715510008  551 - Tecnico em Instrumento Musical Integrado...   
6377  201715510011  551 - Tecnico em Instrumento Musical Integrado...   
6462  201715510012  551 - Tecnico em Instrumento Musical Integrado...   
6432  201715510013  551 - Tecnico em Instrumento Musical Integrado...   
6585  201715510014  551 - Tecnico em Instrumento Musical Integrado...   
6687  201715510019  551 - Tecnico em Instrumento Musical Integrado...   
6730  201715510020  551 - Tecnico em Instrumento Musical Integrado...   
6788  201715510021  551 - Tecnico em Instrumento Musical Integrado...   
6832  201715510022  551 - Tecnico em Instrumento Musical Integrado...   
7041  201715510026  551 - Tecnico em Instrumento Musical Integrado...   
7157  201715510029  551 - Tecnico em Instrumento Musical Integrado...   
7371  201715510034  551 - Tecnico em Instrumento Musical Integrado...   

      Ano de Ingresso       Cor/Raca  \
324              2016         Branca   
273              2016          Preta   
2176             2016         Branca   
206              2016          Parda   
1668             2016          Parda   
1378             2016          Parda   
1230 

In [47]:
del dfs_cadastro_geral_disciplinas_concluidas_primeiro_ano['Codigo do Componente'] 

dfs_cadastro_msi = dfs_cadastro_geral[(dfs_cadastro_geral['Curso'].str.contains("550"))]

In [48]:
dfs_resumo_geral = dfs[['Matricula','Codigo do Componente','Descricao do Componente','Nota']]

In [49]:
dfs_resumo_geral['Nota'] = dfs_resumo_geral['Nota'].replace(['-'],'0')

/usr/local/lib/python3.6/dist-packages/ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  """Entry point for launching an IPython kernel.


dfs_resumo_geral = dfs_resumo_geral[(~dfs_resumo_geral['Situacao no Diario'].str.contains("Cancelado"))]

dfs_resumo_geral = dfs_resumo_geral[(~dfs_resumo_geral['Situacao no Curso'].str.contains("Cancelado"))]

In [50]:
dfs_resumo_geral.head()

Matricula Codigo do Componente Descricao do Componente Nota
0  201615500018             TIN.0003                   Artes   76
1  201615500018             TIN.0022              Biologia I   75
2  201615500018             TIN.0112       Educacao Fisica I   92
3  201615500018             TIN.0086             Filosofia I   88
4  201615500018             TIN.0125                Fisica I   89

In [51]:
dfs_resumo_geral['Nota'] = dfs_resumo_geral['Nota'].astype(int)

/usr/local/lib/python3.6/dist-packages/ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  """Entry point for launching an IPython kernel.


In [52]:
dfs_resumo_geral.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 16204 entries, 0 to 16203
Data columns (total 4 columns):
 #   Column                   Non-Null Count  Dtype 
---  ------                   --------------  ----- 
 0   Matricula                16204 non-null  int64 
 1   Codigo do Componente     16204 non-null  object
 2   Descricao do Componente  16204 non-null  object
 3   Nota                     16204 non-null  int64 
dtypes: int64(2), object(2)
memory usage: 506.5+ KB


dfs_resumo_msi[(dfs_resumo_msi.Matricula == 202015500035)]

In [53]:
dfs_resumo_geral_primeiro_ano = dfs_resumo_geral[(dfs_resumo_geral['Codigo do Componente'].str.contains('|'.join(searchfor_cod_disciplinas1)))]

In [54]:
df_pivot_geral = dfs_resumo_geral_primeiro_ano.pivot_table(index="Matricula", columns=["Descricao do Componente"], values=["Nota"], aggfunc=np.mean)

df_pivot_geral = pd.merge(dfs_cadastro_geral,
                 df_pivot_geral,
                 on='Matricula', 
                 how='left')

In [55]:
dfs_pivot_geral_primeiro_ano = dfs_resumo_geral_primeiro_ano.pivot_table(index="Matricula", columns=["Descricao do Componente"], values=["Nota"], aggfunc=np.mean)

In [56]:
dfs_pivot_geral_primeiro_ano = pd.merge(dfs_cadastro_geral_disciplinas_concluidas_primeiro_ano,
                 dfs_pivot_geral_primeiro_ano,
                 on='Matricula', 
                 how='left')


/home/jp/.local/lib/python3.6/site-packages/pandas/core/reshape/merge.py:643: UserWarning: merging between different levels can give an unintended result (1 levels on the left,2 on the right)
  warnings.warn(msg, UserWarning)


In [57]:
dfs_pivot_geral_primeiro_ano.reset_index(drop=True)

Matricula                                              Curso  \
0    201615500003  550 - Tecnico em Manutencao e Suporte em Infor...   
1    201615500007  550 - Tecnico em Manutencao e Suporte em Infor...   
2    201615500008  550 - Tecnico em Manutencao e Suporte em Infor...   
3    201615500009  550 - Tecnico em Manutencao e Suporte em Infor...   
4    201615500014  550 - Tecnico em Manutencao e Suporte em Infor...   
..            ...                                                ...   
227  201915520039  552 - Tecnico em Edificacoes Integrado ao Ensi...   
228  201915520041  552 - Tecnico em Edificacoes Integrado ao Ensi...   
229  201915520042  552 - Tecnico em Edificacoes Integrado ao Ensi...   
230  201915520044  552 - Tecnico em Edificacoes Integrado ao Ensi...   
231  201915520045  552 - Tecnico em Edificacoes Integrado ao Ensi...   

     Ano de Ingresso Cor/Raca  \
0               2016   Branca   
1               2016    Preta   
2               2016   Branca   
3               2016    Parda   
4               2016    Parda   
..               ...      ...   
227             2019   Branca   
228             2019    Parda   
229             2019   Branca   
230             2019    Parda   
231             2019    Parda   

                                              Cota MEC     Cota SISTEC  \
0                                        Nao se aplica   Nao se aplica   
1                                        Nao se aplica   Nao se aplica   
2                                        Nao se aplica   Nao se aplica   
3                                        Nao se aplica   Nao se aplica   
4    Oriundo de escola publica, com renda inferior ...  Escola Publica   
..                                                 ...             ...   
227                                      Nao se aplica   Nao se aplica   
228                                      Nao se aplica   Nao se aplica   
229                                      Nao se aplica   Nao se aplica   
230  Oriundo de escola publica, com renda inferior ...        Cor/Raca   
231  Oriundo de escola publica, com renda inferior ...        Cor/Raca   

    Data de Nascimento Faixa de Renda (SISTEC)  \
0           01/04/2001              0<RFP<=0,5   
1           18/03/2001           Nao declarado   
2           14/12/2000            1,0<RFP<=1,5   
3           26/11/2001           Nao declarado   
4           25/04/2002           Nao declarado   
..                 ...                     ...   
227         26/03/2004              0<RFP<=0,5   
228         23/03/2004              0<RFP<=0,5   
229         26/05/2003                 RFP>3,5   
230         06/01/2004              0<RFP<=0,5   
231         12/06/2003              0<RFP<=0,5   

                              Nome da Escola de Origem Sexo  ...  \
0                                                    -    M  ...   
1    Escola Agrotecnica de Ensino Fundamental Deput...    M  ...   
2                     Colegio Nossa Senhora de Lourdes    F  ...   
3     Instituto Educacional Jose Pereira do Nascimento    M  ...   
4    Escola Agrotecnica de Ensino Fundamental Deput...    F  ...   
..                                                 ...  ...  ...   
227             E.M.E.F II PROF. MAURO SEVERIANO LEITE    F  ...   
228                   INSTITUTO EDUCACIONAL PAI ETERNO    M  ...   
229   INSTITUTO EDUCACIONAL JOSE PEREIRA DO NASCIMENTO    M  ...   
230        EMEF II PROFESSORA MARIA  DO SOCORRO ARAGAO    M  ...   
231            ESCOLA MUNICIPAL MARIA BEZERRA DA SILVA    M  ...   

    (Nota, Historia da Musica I) (Nota, Informatica)  \
0                            NaN                 NaN   
1                            NaN                 NaN   
2                            NaN                 NaN   
3                            NaN                 NaN   
4                            NaN                 NaN   
..                           ...                 ...   
227                          NaN               

In [58]:
dfs_pivot_geral_primeiro_ano.to_csv('suap_geral_primeiro_ano.csv', encoding='utf-8',sep=';')